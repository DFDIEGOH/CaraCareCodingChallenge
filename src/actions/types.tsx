/* eslint-disable prettier/prettier */
export const SEARCH_MOVIE = 'SEARCH_MOVIE';
export const SELECT_MOVIE = 'SELECT_MOVIE';
