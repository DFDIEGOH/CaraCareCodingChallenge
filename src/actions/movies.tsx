import axios from 'axios';
import {SEARCH_MOVIE, SELECT_MOVIE} from './types';

const path = 'http://www.omdbapi.com/?i=tt3896198&apikey=a120ef66';

export function searchMovies(searchParam) {
  return (dispatch) => {
    axios.get(path + '&s=' + searchParam).then((res) => {
      dispatch({type: SEARCH_MOVIE, payload: res.data.Search});
    });
  };
}
