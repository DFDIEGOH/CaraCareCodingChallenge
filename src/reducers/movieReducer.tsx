/* eslint-disable prettier/prettier */
import {SEARCH_MOVIE, SELECT_MOVIE} from '../actions/types';

const initialState = {
  search: 'Search...(state)',
  results: [],
  selected: {},
};

const movieReducer = (state = initialState, action) => {
  switch (action.type) {
    case SEARCH_MOVIE:
      return {
        ...state,
        results: action.payload,
      };
    case SELECT_MOVIE:
      return {
        ...state,
        selected: action.payload,
      };

    default:
      return state;
  }
};

export default movieReducer;
