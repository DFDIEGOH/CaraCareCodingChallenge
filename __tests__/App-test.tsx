/**
 * @format
 */

import 'react-native';
import axios from 'axios';

import * as actions from '../src/actions/movies';
import * as types from '../src/actions/types';

import reducer from '../src/reducers/movieReducer';
import {initialState} from '../src/reducers/movieReducer';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const pathApi = 'http://www.omdbapi.com/?i=tt3896198&apikey=a120ef66';

describe('API calls', () => {
  it('Api Call - Contains Elements', () => {
    axios.get(pathApi).then((res) => {
      let results = res.data;
      expect(results).toContain(results.Title);
    });
  });

  it('Api Call by Title', () => {
    axios.get(pathApi + '&t=Cars').then((res) => {
      let results = res.data;
      expect(results.Title).toBe('Cars');
    });
  });
});

describe('Reducers', () => {
  it('Should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('Should handle SEARCH_MOVIE', () => {
    expect(
      reducer(initialState, {
        type: 'SEARCH_MOVIE',
      }),
    ).toMatchSnapshot();
  });
});
