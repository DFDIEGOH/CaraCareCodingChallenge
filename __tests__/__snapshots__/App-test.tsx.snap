// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers should handle TODO_REQUEST 1`] = `
Object {
  "results": undefined,
  "search": "Search...(state)",
  "selected": Object {},
}
`;

exports[`reducers should return the initial state 1`] = `
Object {
  "results": Array [],
  "search": "Search...(state)",
  "selected": Object {},
}
`;
